File: src/main/java/com/alibou/book/book/BorrowedBookResponse.java
package com.alibou.book.book;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class BorrowedBookResponse {

    private Long id;
    private String title;
    private String authorName;
    private String isbn;
    private double rate;
    private boolean returned;
    private boolean returnApproved;
    private Genre genre;  // Added genre field

    // NEW: Added date fields for tracking
    private LocalDateTime borrowedDate;
    private LocalDateTime returnedDate;
}

File: book-network-ui/src/app/services/models/borrowed-book-response.ts
/* tslint:disable */
/* eslint-disable */
export interface BorrowedBookResponse {
  authorName?: string;
  id?: number;
  isbn?: string;
  rate?: number;
  returnApproved?: boolean;
  returned?: boolean;
  title?: string;
  // NEW: Added date fields for tracking
  borrowedDate?: string;
  returnedDate?: string;
}
